examples/pydantic_ai_examples/lancedb_multimodal.py

```diff
@@ -0,0 +1,277 @@
+"""Multimodal RAG with LanceDB.
+
+What this does:
+- Fetches a product catalog from a live API (fakestoreapi.com).
+- Embeds product descriptions using a CLIP model (clip-ViT-B-32) and stores them in LanceDB.
+- Implements a RAG agent with a `find_products` tool that can:
+    1. Perform semantic search (e.g., "something for the cold weather").
+    2. Perform metadata filtering (e.g., "show me all electronics").
+       (e.g., "a cool t-shirt" from the "men's clothing" category under $20).
+- Generates and displays an image collage of the results for instant visual feedback.
+- Creates logfire tracing dashboard if api key is set
+
+Install dependencies:
+    pip install lancedb sentence-transformers torch httpx pandas Pillow logfire[httpx]
+
+Set your Google API key (for the agent's text generation):
+    export GOOGLE_API_KEY=your_api_key_here
+
+Usage:
+    # First, build the product database from the live API
+    python lancedb_multimodal.py build
+
+    # Then, ask for a recommendation:
+    python lancedb_multimodal.py search "a cool t-shirt in men's clothing under 20 dollars"
+"""
+
+import asyncio
+import io
+import os
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional, cast
+from pydantic_ai import Agent, RunContext
+
+import httpx
+
+try:
+    import lancedb
+    import logfire
+    from lancedb.pydantic import LanceModel, Vector
+    from PIL import Image
+    from sentence_transformers import SentenceTransformer
+except ImportError:
+    print(
+        """Missing dependencies. To run this example, please install the required packages by running:
+pip install lancedb sentence-transformers torch httpx pandas Pillow logfire[httpx]"""
+    )
+    sys.exit(0)
+
+
+# ruff: noqa
+# pyright: reportMissingImports=false
+# pyright: reportInvalidTypeForm=false
+# pyright: reportUnknownVariableType=false
+# pyright: reportUnknownMemberType=false
+# pyright: reportUnknownArgumentType=false
+# pyright: reportUntypedBaseClass=false
```

Comment:
Why so many?

Reply:
to be clear, we're happy to accept an example from you, but this is just a big middle finger to how we operate.

it's either saying:
* I don't care about any of your typing, linting, DX work
* or, we're special, we don't need to do this

Either lanceDB is typesafe, in which case remove these. or LanceDB is not type-safe in which case the example probably doesn't belong in our docs!

Reply:
Added these to figure out the deps part while fighting CI.. Will revert back